cog/core/sql_abstract.py

```
@@ -0,0 +1,187 @@
+
+        if is_protected and is_init:
+            raise AttributeError(
+                f"The protected attribute `{name}` should be read-only after initialization."
+            )
+        return super().__setattr__(name, value)
+
+
+class SQLTable(ABC):
+    @staticmethod
+    @abstractmethod
+    def from_sql(unique_id): ...
```

Ruff (0.8.2)

Missing return type annotation for staticmethod `from_sql`

(ANN205)